### PR TITLE
[cli] Add custom user agent for CLI

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
This puts the CLI version in the User agent of the CLI

### Test Plan
I used a modified local testnet node that prints out the user agent of every request:

```
# Browser
Agent: Some("Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36")

# Curl
Agent: Some("curl/7.85.0")

# CLI
Agent: Some("aptos-cli / 1.0.7")
```

### CODE REVIEW

**Review by 0xmaayan — APPROVED** [review]

File: crates/aptos/src/common/types.rs [hunk 1]
```diff
@@ -48,6 +48,7 @@ use std::{
 };
 use thiserror::Error;
 
+pub const USER_AGENT: &str = concat!("aptos-cli / ", env!("CARGO_PKG_VERSION"));
```
Comment (banool) [inline crates/aptos/src/common/types.rs:51]: I see in your previous examples there is no space on each side of the slash. Can we do that here too?
```suggestion
pub const USER_AGENT: &str = concat!("aptos-cli/", env!("CARGO_PKG_VERSION"));
```

**Comment by banool** [comment]
Another problem I see with this approach is it makes the parsing harder. As in, we'll have to figure out how to identify our own part of the user agent string. If we always mark our own user agent as `aptos-*` it should be possible though. Do you have any thoughts on this?

**Review by davidiw — APPROVED** [review]

**Comment by banool** [comment]
Let's figure out the solution to how we'll log this before merging.

**Comment by gregnazario** [comment]
> Let's figure out the solution to how we'll log this before merging.

This net doesn't change anything except for how messages will be sent by the CLI.  Given this needs to be in the packaged software earlier than later, I'm merging it in.

**Comment by banool** [comment]
Cancelling auto merge again, see message in Slack.

File: crates/aptos/src/common/types.rs (hunk 1 above)
Comment (gregnazario) [reply crates/aptos/src/common/types.rs]: Yep, David removed the slash on both sides now which is the de facto way that it is done.

**Pull request merged**